cg/services/analysis_starter/configurator/implementations/balsamic.py

```diff
@@ -0,0 +1,306 @@
+        self.ponn_directory: Path = Path(config.balsamic.ponn_directory)
+        self.slurm_account: str = config.slurm.account
+        self.slurm_mail_user: str = config.slurm.mail_user
+        self.swen_snv: str = config.swegen_snv
+        self.swen_sv: str = config.swegen_sv
+
+    def configure(self, case_id: str, **flags) -> BalsamicCaseConfig:
+        config_cli_input: BalsamicConfigInput = self._build_cli_input(case_id)
+        self.create_config_file(config_cli_input)
+        return self.get_config(case_id=case_id, **flags)
+
+    def get_config(self, case_id: str, **flags) -> BalsamicCaseConfig:
+        balsamic_config: BalsamicCaseConfig = BalsamicCaseConfig(
+            account=self.slurm_account,
+            binary=self.balsamic_binary,
+            conda_binary=self.conda_binary,
+            cluster_config=self.default_cluster_config,
+            environment=self.balsamic_binary,
+            mail_user=self.slurm_mail_user,
+            qos=self.store.get_case_by_internal_id(case_id).slurm_priority,
+            sample_config=self._get_sample_config_path(case_id),
+        )
+        balsamic_config: BalsamicCaseConfig = self._set_flags(config=balsamic_config, **flags)
+        self._ensure_valid_config(balsamic_config)
+        return balsamic_config
+
+    def _build_cli_input(self, case_id) -> BalsamicConfigInput:
+        case: Case = self.store.get_case_by_internal_id(case_id)
+        if self._all_samples_are_wgs(case):
+            return self._build_wgs_config(case)
+        else:
+            return self._build_targeted_config(case)
+
+    def _build_wgs_config(self, case: Case) -> BalsamicConfigInput:
+        patient_sex: SexOptions = self._get_patient_sex(case)
+        return BalsamicConfigInput(
+            analysis_dir=self.root_dir,
+            analysis_workflow=case.data_analysis,
+            artefact_snv_observations=self.loqusdb_artefact_snv,
+            balsamic_binary=self.balsamic_binary,
+            balsamic_cache=self.cache_dir,
+            cadd_annotations=self.cadd_path,
+            cancer_germline_snv_observations=self.loqusdb_cancer_germline_snv,
+            cancer_germline_sv_observations=self.loqusdb_cancer_germline_sv,
+            cancer_somatic_snv_observations=self.loqusdb_cancer_somatic_snv,
+            cancer_somatic_sv_observations=self.loqusdb_cancer_somatic_sv,
+            case_id=case.internal_id,
+            clinical_snv_observations=self.loqusdb_clinical_snv,
+            clinical_sv_observations=self.loqusdb_clinical_sv,
+            conda_binary=self.conda_binary,
+            fastq_path=Path(self.root_dir, case.internal_id, "fastq"),
+            gender=patient_sex,
+            genome_interval=self.genome_interval_path,
+            genome_version=GenomeVersion.HG19,
+            gens_coverage_pon=(
+                self.gens_coverage_female_path
+                if patient_sex == SexOptions.FEMALE
+                else self.gens_coverage_male_path
+            ),
```

Comment:
Is this clear or should it be a small method?

Reply:
small method 👍